kotsadm/pkg/handlers/redact.go

```diff
@@ -159,3 +187,271 @@ func GetRedact(w http.ResponseWriter, r *http.Request) {
 	getRedactResponse.UpdatedSpec = data
 	JSON(w, 200, getRedactResponse)
 }
+
+func GetRedactMetadata(w http.ResponseWriter, r *http.Request) {
+	w.Header().Set("Access-Control-Allow-Origin", "*")
+	w.Header().Set("Access-Control-Allow-Headers", "content-type, origin, accept, authorization")
+
+	if r.Method == "OPTIONS" {
+		w.WriteHeader(200)
+		return
+	}
+
+	getMetadataResponse := RedactorMetadataResponse{
+		Success: false,
+	}
+
+	sess, err := session.Parse(r.Header.Get("Authorization"))
+	if err != nil {
+		logger.Error(err)
+		getMetadataResponse.Error = "failed to parse authorization header"
+		JSON(w, 401, getMetadataResponse)
+		return
+	}
+
+	// we don't currently have roles, all valid tokens are valid sessions
+	if sess == nil || sess.ID == "" {
+		getMetadataResponse.Error = "failed to parse authorization header"
```

Comment:
maybe a different message here, because this one is already used

Reply:
sounds like a good idea